src/libmongoc/src/mongoc/mongoc-server-description.c

```diff
@@ -807,6 +807,9 @@ mongoc_server_description_new_copy (
                                               &description->error);
    } else {
       mongoc_server_description_reset (copy);
+      /* preserve the original server description type, which is manually set
```

Comment:
Note, this was not necessary for regular (non load balanced) server descriptions, which determine the type when processing the last `last_hello_response`.

But a load balanced server description does not have a hello response. It is manually created.